File: ChallengeBackend/src/main/java/br/com/savedra/challengebackend/repository/MessageRepository.java
package br.com.savedra.challengebackend.repository;

import br.com.savedra.challengebackend.model.Message;
import org.springframework.data.mongodb.repository.MongoRepository;
import java.util.List;

public interface MessageRepository extends MongoRepository<Message, String> {
    List<Message> findByChatRoomIdOrderByTimestampAsc(String chatRoomId);
}

File: app/src/main/java/br/com/savedra/challengecrm/data/repository/MessageRepository.kt
package br.com.savedra.challengecrm.data.repository

import br.com.savedra.challengecrm.model.Message
import com.google.firebase.firestore.FirebaseFirestore

class MessageRepository {

  private val db = FirebaseFirestore.getInstance()

  fun sendMessage(message: Message, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
    db.collection("messages")
      .add(message)
      .addOnSuccessListener {
        onSuccess()
      }
      .addOnFailureListener { e ->
        onFailure(e)
      }
  }
}